python/triton/language/core.py

```diff
@@ -1498,10 +1499,21 @@ def dot(input, other, acc=None, input_precision=None, max_num_imprecise_acc=None
     :type other: 2D tensor of scalar-type in {:code:`float16`, :code:`bfloat16`, :code:`float32`}
     :param input_precision: How to exercise the Tenors cores for f32 x f32. If the device does not have Tensor Cores
         or the inputs are not of dtype f32, this option is ignored.
+    :param allow_tf32: *Deprecated.*  If true, input_precision is set to "tf32".
+      Only one of :code:`input_precision` and :code:`allow_tf32` can be
+      specified (i.e. at least one must be :code:`None`).
     :type other: string. Available options for nvidia: :code:`"tf32"`, :code:`"tf32x3"`, :code:`"ieee"`. Default: :code:`"tf32"`. Avaliable options for amd: :code:`"ieee"`.
     """
+    assert input_precision is None or allow_tf32 is None, "Only one of input_precision and allow_tf32 can be specified"
+    if allow_tf32 is None:
+        allow_tf32 = True
+
     if input_precision is None:
-        input_precision = os.getenv("TRITON_F32_DEFAULT", None)
+        if allow_tf32 is not None:
```

Comment:
allow_tf32 is never None here based on the code above?

Reply:
Oops, thanks, fixed.

Reply:
so the else still cannot be reached right?

Reply:
Ugh, thank you.  How about this?